src/Meshes/trailMesh.ts

```diff
@@ -154,8 +153,8 @@ export class TrailMesh extends Mesh {
             let alpha: number = 2 * Math.PI / this._sectionPolygonPointsCount;
             for (let i: number = 0; i < this._sectionPolygonPointsCount; i++) {
                 this._sectionVectors[i].copyFromFloats(
-                    Math.cos(i * alpha) * this._diameter,
-                    Math.sin(i * alpha) * this._diameter,
+                    Math.cos(i * alpha) * this._dimensions[0],
+                    Math.sin(i * alpha) * this._dimensions[1],
                     0
                 );
                 this._sectionNormalVectors[i].copyFromFloats(

```

Comment:
When changing line 161 from 
```
Math.cos(i * alpha),
```
to 
```
0,
```

It avoids it poking through on this playground https://playground.babylonjs.com/#KXWZPA but you can see the black issue described in the other thread. Toggling backfaceculling didnt seem to work.

Reply:
The normals are tricky, even using computeNormals doesn't help. Setting the material's emissive color works well, but without it the normals don't look right. Flipping them doesn't help either.

changing line 147
```
positions[i - 3 * this._sectionPolygonPointsCount] = positions[i] - normals[i] / this._length * (this._dimensions[0] > this._dimensions[1] ? this._dimensions[0] : this._dimensions[1]);
```
to
```
positions[i - 3 * this._sectionPolygonPointsCount] = positions[i] - normals[i] / this._length / (this._dimensions[0] > this._dimensions[1] ? this._dimensions[0] : this._dimensions[1]);
```
the trail doesn't taper at the end, but it looks cleaner

https://brave-gates-959b79.netlify.com/

Reply:
Interesting, I think a tapering end should still be supported. I'll keep playing with the code unless you have any other ideas.

Reply:
Should I close that PR or do you think we will work on it more?

Reply:
I'm fine with this being closed I ended up using spheres for a demo instead of uneven objects.
![ezgif-3-50bd3ebf7ed5](https://user-images.githubusercontent.com/3753071/55916389-17487e80-5ba1-11e9-8ad3-a33c7c58666a.gif)
so unless @danjpar feels additional modifications are required im good.

Reply:
You can go ahead and close this. I will keep trying to improve it after I finish my project and get some free time. Might be a couple weeks.